### PR TITLE
Changes required to let scriptrunner.py run commands outside of Helix Context

### PR DESCRIPTION
**Pull request by JohnTortugo**
- Removed the parameter "--standalone"

- Added a few checks to prevent problems when some Helix environment variables aren't set.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py [hunk 1]
```diff
@@ -26,27 +26,23 @@ def _main(settings, optlist, args):
...
-
         execution_args = [os.path.join(unpack_dir, script_to_execute)] + args
 
+        test_executor = HelixTestExecution(settings)
+
         return_code = helix.proc.run_and_log_output(
-            execution_args,
-            cwd=unpack_dir,
-            env=None
+       	    execution_args,
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: mixing tabs and spaces all over, please unify to spaces.

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py [hunk 2]
```diff
@@ -26,27 +26,23 @@ def _main(settings, optlist, args):
...
-                --script
-                [--args arg1 arg2...]
+                --script=path
+                [args]
         """
-        optdict = dict(optlist)
         log.info("BuildTools Helix Script Runner v0.1 starting")
-        if '--args' in optdict:
-            script_arguments = optdict['--args']
-            log.info("Script Arguments:"+script_arguments)
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: any special reason to delete a useful log statement?

Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: Thanks Matt. My intent was to simplify the code by removing an if+variable that was only used for logging. I had then changed the code to dump the script_arguments in another place; However, after the recent changes I had also stripped that off. I'll move the log back.

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py (hunk 1 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: Will fix it.

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py [hunk 3]
```diff
@@ -71,21 +67,26 @@ def _main(settings, optlist, args):
                             test_count = int(match.groups()[0])
                         break
 
-            result_url = test_executor.upload_file_to_storage(results_location, settings)
+            if settings.output_uri is not None:
+                result_url = test_executor.upload_file_to_storage(results_location, settings)
+            else:
+                result_url = None;
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: You probably know python better than I, is this else clause actually doing anything?  (as I could access result_url before and it'd be None then too)

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py [hunk 4]
```diff
@@ -26,27 +26,23 @@ def _main(settings, optlist, args):
...
+
         return_code = helix.proc.run_and_log_output(
-            execution_args,
-            cwd=unpack_dir,
-            env=None
+       	    execution_args,
+       	    cwd=unpack_dir,
+       	    env=None
         )
         event_client = helix.event.create_from_uri(settings.event_uri)
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: When running in a repro environment, we don't want the machine sending telemetry at all and will have not provided a value for the event URI in settings (if things went correctly...).  All the event_client usages here should be conditional on the presence or absence of an event_uri.

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py [hunk 5]
```diff
@@ -71,21 +67,26 @@ def _main(settings, optlist, args):
...
                         break
 
-            result_url = test_executor.upload_file_to_storage(results_location, settings)
+            if settings.output_uri is not None:
+                result_url = test_executor.upload_file_to_storage(results_location, settings)
+            else:
+                result_url = None;
+
             log.info("Sending completion event")
             event_client.send(
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: This whole section shouldn't occur unless there's a valid event uri.

**Comment by MattGal** [comment]
Changes are mostly reasonable with some noted changes I'd like to see.  Have you tested this in the normal execution flow?

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py (hunk 3 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: Well, I am not an expert in Python but just tested and without the assignment in the else clause I got an `Unhandled error: local variable 'result_url' referenced before assignment` in the next lines.

File: src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py [hunk 6]
```diff
@@ -71,21 +69,27 @@ def _main(settings, optlist, args):
...
-                    'CorrelationId': settings.correlation_id,
-                    'ResultsXmlUri': result_url,
-                    'TestCount': test_count,
-                }
+            if settings.output_uri is not None:
+                result_url = test_executor.upload_file_to_storage(results_location, settings)
+            else:
+                result_url = None;
+
+            if (settings.event_uri is not None):
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: this should be gated on whether event_client is null and not create one earlier if the event_uri is not available.

Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: What do you think about moving the creation of event_client to the bottom and so we only need one condition?

```
            if (settings.event_uri is not None):
                event_client = helix.event.create_from_uri(settings.event_uri);
                log.info("Sending completion event")
                event_client.send(
                    {
                        'Type': 'XUnitTestResult',
                        'WorkItemId': settings.workitem_id,
                        'WorkItemFriendlyName': settings.workitem_friendly_name,
                        'CorrelationId': settings.correlation_id,
                        'ResultsXmlUri': result_url,
                        'TestCount': test_count,
                    }
                )
```

Comment (MattGal) [reply src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/scriptrunner/scriptrunner.py]: totally fine.

**Pull request merged**